src/coreclr/vm/reflectioninvocation.cpp

```diff
@@ -823,7 +823,7 @@ FCIMPL5(Object*, RuntimeMethodHandle::InvokeMethod,
 
     LPBYTE pTransitionBlock = pAlloc + TransitionBlock::GetNegSpaceSize();
 
-    CallDescrData callDescrData;
+    CallDescrData callDescrData = {};
```

Comment:
The `callDescrData.returnValue`, which is a complex type anyways. I'll need to call `memset()` I think.

Reply:
Is there a better way to tell the compiler to not worry about initializating something? `returnValue` gets initialized by the assembly helper that this calls.

Reply:
I don't think so, but I can consolidate the uses and make a comment on why. Please take a look at the mitigation.

Reply:
What is the build flavor and VS version that hits this warning? I have tried to build locally to understand what's going on, but I am not able to reproduce the warning. I have tried x64 release and x64 checked on current VS dogfood.

Reply:
I am building using Debug x64. My VS version is Version 17.2.0 Preview 3.0 [32311.354.main].

The issue is all fields except the `returnValue` are explicitly set. That field is used at the end of the method and the analysis is indicating it is used without being initialized.

Reply:
I think that the compiler is getting confused by the EH macros. Based on my experiments, you can workaround the warning by adding a dummy assignment to the exception handling path like this:

```
        // Abuse retval to store the exception object
        gc.retVal = GET_THROWABLE();
        _ASSERTE(gc.retVal);

#ifdef _MSC_VER
        // Workaround bogus MSVC warning about uninitialized local variables
        *(BYTE*)&callDescrData.returnValue = 0;
#endif

        fExceptionThrown = true;
```
This will make the warning go away (at the cost of adding one extra instruction on a cold path), without adding a suppression of the warning that will need to be audited for each release.